test/protocol.js

```diff
@@ -153,6 +153,11 @@ describe("Protocol", () => {
 
                 tests.forEach(function (test) {
                     it(`should store ${test.ext} data with a (${test.cmd}) command (client write packet size = ${test.packetSize})`, () => {
+                        // Insert 'q' character ('Quit' command) into the GUID, to catch subtle protocol errors when packet size is 1
+                        if(test.packetSize === 1) {
+                            test.data.guid[test.data.guid.length - 1] = 'q'.charCodeAt(0);
```

Comment:
I wouldn't expect it to, but just in case I'll add it to the test. It shouldn't matter positionally where the q is encountered .. as long as it is somewhere after a 2 char command is received, it would have triggered the bug.